src/arrayMethodJoin.js

```diff
@@ -4,9 +4,25 @@
  * Implement method join
  */
 function applyCustomJoin() {
-  [].__proto__.join2 = function(separator) {
-    // write code here
+  [].__proto__.join2 = function(separator = ',') {
+    let resultJoin = '';
+
+    if (this.length === 0) {
+      return resultJoin;
+    }
+
+    for (let i = 0; i < this.length; i++) {
+      if (this[i] === undefined || this[i] === null || this[i] === ' ') {
+        resultJoin += separator;
+      } else if (i === this.length - 1) {
+        resultJoin += this[i];
+      } else {
+        resultJoin += this[i] + `${separator}`;
+      }
+    }
```

Comment:
```suggestion
    for (let i = 0; i < this.length; i++) {
      if (this[i] !== undefined || this[i] !== null || this[i] !== ' ') {
        resultJoin += this[i];
      }
      if (i !== this.length - 1) {
        resultJoin += separator
      }
    }
```
You can simplify your code and make 2 checks instead of 3.
In the case I've described, you will always add `this[i]` to the new string when `this[i] !==null/undefined` (when this[i] is null or undefined the check won't pass, and will happen nothing). Also, you always adding separator while `i !== this.length - 1`